src/core.ts

```diff
@@ -337,13 +280,13 @@ export async function executeHipthrustable<
   };
 
   try {
-    // @todo: allow doWork to return more context instead of "true" too.  Anything falsy will be interpreted as an error.  Don't forget || {} after call
-    if (requestHandler.doWork) {
-      // to keep executeHipthrustable from being too opinionated, it's doWork's responsibility to handle and throw client errors.
-      // Any un-boom'ed errors here should be interpreted as server errors
-      await Promise.resolve(requestHandler.doWork(finalAuthContext));
-    }
-    const doWorkContext = finalAuthContext;
+    // to keep executeHipthrustable from being too opinionated, it's doWork's responsibility to handle and throw client errors.
+    // Any un-boom'ed errors here should be interpreted as server errors
+    const doWorkContextOnly = await Promise.resolve(
+      requestHandler.doWork(finalAuthContext)
+    );
+
+    const doWorkContext = { ...finalAuthContext, ...doWorkContextOnly };
 
```

Comment:
that comment you deleted should have been followed:
1. doWork could have returned true - in that case, this code would fail.
2. it should be considered a failure if it returns anything other than "true" or a non-empty object
I think we may have done it right in HTPipe - look there
- [ ] fix 2 bugs with doWork

Reply:
as discussed
doWork can return only void or object
i don't see why it should be considered a failure when it returns anything other then "true" or non-empty object because this is doWork function
it can return even void
if doWork throw error we will handle it
also i add || {}; in case if we will return void it will become {} so we can use it in doWorkContext object